flake.nix

```diff
@@ -1,30 +1,207 @@
+      # inputs.nixpkgs.follows = "nixpkgs"; # TODO: should we follow this?
     };
+
+    nft-marketplace-frontend.url = github:mlabs-haskell/nft-marketplace/aciceri/nix; # TODO: change to master once this is merged: https://github.com/mlabs-haskell/nft-marketplace/pull/224
+
+    nft-marketplace-server = {
+      url = github:mlabs-haskell/nft-marketplace-server/3bcb7606b9172e17c9c70961ad1b083130430fee;
+      # flake = false;
+    };
+
+    # nft-marketplace-server needs some specific versions.
+    nixpkgs-nft-marketplace-server.url = github:nixos/nixpkgs/2cf9db0e3d45b9d00f16f2836cb1297bcadc475e;
+
+    ogmios-datum-cache = {
+      url = github:mlabs-haskell/ogmios-datum-cache/f8c671aebeb84d57b4879532073e20f8567c5ed4;
+      # inputs.nixpkgs.follows = "nixpkgs"; # TODO: should we follow this?
+      # flake = false;
+    };
+
+    seabug-contracts.url = "github:mlabs-haskell/seabug-contracts/23f49cf05d6230a8c1f63924ac9e61d1e1c0d5a8";
   };
-  outputs = { self, nixpkgs, cardano-node, flake-utils, arion }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let pkgs = nixpkgs.legacyPackages.${system};
-      in rec {
-        devShell = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; [
-            arion.packages.${system}.arion
-            cardano-node.packages.${system}.cardano-cli
-            curl
-            httpie
-            ipfs
-            jq
-            postgresql
-            shfmt
-            expect
+  outputs =
+    { self
+    , nixpkgs
+    , cardano-node
+    , arion
+    , nft-marketplace-frontend
+    , nft-marketplace-server
+    , ogmios-datum-cache
+    , cardano-transaction-lib
+    , ...
+    } @ inputs:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+
+      perSystem = nixpkgs.lib.genAttrs supportedSystems;
+
+      overlays.default = final: prev:
+        let
+          inherit (final) system;
+        in
+        {
+          nft-marketplace-server = nft-marketplace-server.packages.${system}."nft-marketplace-server:exe:nft-marketplace-server";
+
+
+          # TODO: system-agnostic
+          inherit (inputs.ogmios-datum-cache.packages.${system}) ogmios-datum-cache;
+
+          cardano-transaction-lib-server =
+            # cardano-transaction-lib.packages.${system}."cardano-browser-tx-server:exe:cardano-browser-tx-server";
+            cardano-transaction-lib.packages.${system}."ctl-server:exe:ctl-server";
+
+          nft-marketplace-frontend-artifacts = nft-marketplace-frontend.packages.${system}.default;
+        };
+
+      pkgsFor = system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ overlays.default ];
+        };
+
+      nixosModules.default = import ./seabug-module.nix {
+        arionModule = arion.nixosModules.arion;
+        seabugOverlay = overlays.default;
+      };
+
+      # Build with `nix build nixosConfigurations.seabug.config.system.build.toplevel`
+      # Run with ./result
+      nixosConfigurations.seabug-vm =
```

Comment:
What is the VM stuff? And also `nixosModules`? I'm unclear on what it's for or what it does. I think we should add some documentation on it

Reply:
Mmmm, TBH I don't know how useful it is, I saw that in the previous PR (that I reprise to add the frontend input) there was a [Nixos system](https://github.com/mlabs-haskell/seabug/blob/8e6466611d3b164917aac5a5071daf25f448506d/flake.nix#L92) exported so I simply added a script to run it, then @Synthetica9 expanded and improved it.